service/src/main/kotlin/fi/espoo/evaka/placement/PlacementService.kt

```diff
@@ -352,6 +353,101 @@ private fun handleFiveYearOldDaycare(tx: Database.Transaction, childId: UUID, ty
     )
 }
 
+fun getMissingGroupPlacements(
+    tx: Database.Read,
+    unitId: UUID
+): List<MissingGroupPlacement> {
+    data class GroupPlacementGap(
+        val backup: Boolean,
+        val placementId: UUID,
+        val placementType: PlacementType?,
+        val placementRange: FiniteDateRange,
+        val childId: UUID,
+        val gapRange: FiniteDateRange
+    )
+
+    data class PlacementResult(
+        val id: UUID,
+        val type: PlacementType,
+        val range: FiniteDateRange,
+        val childId: UUID,
+        val groupPlacementRanges: List<FiniteDateRange>
+    )
+    val placements = tx.createQuery(
+        """
+        SELECT 
+            FALSE AS backup,
+            pl.id, 
+            pl.type,
+            daterange(pl.start_date, pl.end_date, '[]') AS range,
+            pl.child_id,
+            array_remove(array_agg(
+                case when dgp.id is null then null else daterange(dgp.start_date, dgp.end_date, '[]') end
+            ), null) AS group_placement_ranges
```

Comment:
is there a better way to avoid zero group placements resulting in an array which contains daterange(null, null)?

Reply:
One option is to use aggregate function filering (`FILTER (WHERE)`):

```
array_agg(...) FILTER (WHERE dgp is NOT NULL)
```

Reply:
excellent!